### ISSUE TITLE
lint to remove unused imports

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#114]
See https://github.com/angular/ts2dart/issues/223

**Comment by pq** [comment]
Interesting.  This check is provided by the `analyzer` in the form of a "hint".

It sounds like you'd like to be able to elevate that to a lint (to use, for example, in a presubmit step)?

**Comment by bwilkerson** [comment]
I suspect that there's a misconception here that the linter will "auto-fix" some number of problems. I don't think that's the current intent for the linter.

I also believe that auto-fix utilities tend to be dangerous. Unless you can verify that there are no false positives and that there is only one reasonable way to fix a problem, you risk breaking the code.

**Comment by srawlins** [comment]
Yeah, unused_import is a solved analyzer hint.

**Comment by thosakwe** [comment]
Is there some functionality in the IDEA plugin that automatically removes unused imports? For as long as I can remember, unused imports in Flutter are removed whenever I press Command+Alt+L to format the document. It used to leave my non-Flutter Dart projects alone, but out of nowhere, now calling the formatter removes imports, which is a frustration when formatting a document I'm not finished writing yet.

Just trying to figure out what the potential causes could be.

**Comment by pq** [comment]
Hey @thosakwe.  Take a look at your Flutter preferences.

![image](https://user-images.githubusercontent.com/67586/43732024-40260538-9965-11e8-8044-7a888f0dd3bc.png)


Maybe you have "Organize imports on save" checked?

**Comment by thosakwe** [comment]
I probably do; thanks for the tip, @pq!

**Comment by xinthink** [comment]
I solved this by [customizing analysis rules](https://dart.dev/guides/language/analysis-options#customizing-analysis-rules) using the `analysis_options.yaml` file:

```yaml
analyzer:
  errors:
    unused_import: error
```

<img width="586" alt="截屏2020-05-24 下午12 53 03" src="https://user-images.githubusercontent.com/390805/82746017-ab819f00-9dbd-11ea-9a5e-4b4f6c38916a.png">
